src/main/default/lwc/lookup/lookup.js

```diff
@@ -112,7 +118,8 @@ export default class Lookup extends LightningElement {
 
         // Compare clean new search term with current one and abort if identical
         const newCleanSearchTerm = newSearchTerm.trim().replace(/\*/g, '').toLowerCase();
-        if (this._cleanSearchTerm === newCleanSearchTerm) {
+
+        if (!this.dispatchSearchEventOnEachRequest && this._cleanSearchTerm === newCleanSearchTerm) {
```

Comment:
I think this logic regarding dispatchSearchEventOnEachRequest should not be implemented in this component but on upper componenent that is using it. What is purpose of this? To avoid to many events?